gpkit/nomials/nomial_math.py

```diff
@@ -739,9 +739,51 @@ def as_posyslt1(self):
 
     def as_gpconstr(self, x0):
         "Returns GP approximation of an SP constraint at x0"
+        print "x0", x0
         siglt0, = self.unsubbed
         x0 = self._fill_default_x0(x0, siglt0.varlocs)
         posy, negy = siglt0.posy_negy()
         mec = (posy.mono_lower_bound(x0) == negy.mono_lower_bound(x0))
         mec.substitutions = self.substitutions
         return mec
+
+    def process_result(self, result):
+        "Checks that all constraints are satisfied with equality"
+        leftsubbed = self.left.sub(result["variables"]).value
+        rightsubbed = self.right.sub(result["variables"]).value
+        log_diff = abs(np.log10(mag(leftsubbed)) - np.log10(mag(rightsubbed)))
+        previously = result.get("sigeq_logsumerror", 0)
+        result["sigeq_logsumerror"] = previously + log_diff
+
+
+class SignomialEqualityTrust(SignomialEquality):
+    """A constraint of the form posynomial == posynomial with trust regions
+
+    The trust regions bound all varkeys in the constraint to stay within
+    a certain region. TODO: only do this for all varkeys?
+    """
+
+    def __init__(self, left, right, trustregion):
+        if trustregion >= 1 or trustregion <= 0:
+            raise ValueError("trust region must be between one and zero.")
+        self.trustregion = float(trustregion)
```

Comment:
maybe the float cast should come before the 0 <= trustregion <= 1 conditional above -- will cause a more informative error for non-float-castable trustregions.